docs/developer-environment-setup.md

```diff
@@ -0,0 +1,82 @@
+5. Run `make install_poetry_deps` to install all python dependencies.
+6. Run `make start_pg` to start the postgres server if it is not already running.
+7. Run `make create_db` to create the database.
+8. Run `make migrate` to run all database migrations.
+9. Run `make run`, which will start running the development server locally (default port is :8000; terminal output will indicate what localhost address to use to access the backend server).
+   - to test it out, navigate to the `/api/docs` endpoint in your browser, you should see documentation for the /add endpoint
+   - navigate to the `api/add?a=4&b=2` endpoint in your browser, which should return as a result the sum of the specified values for a and b.
+10. Optional: to test the Django server's connection to your database, run `python3 manage.py check --database default`
+
+## Backend Environment Use
+
+After doing the initial setup, to get the backend re-running:
+
+1. From the `cas-registration/bc_obps` directory, run `poetry shell`
+2. To set up the database:
+   - If you want to drop and recreate the database with mock data, run `make reset_db`. (Warning: This will delete superusers and you will have to recreate with `make superuser`.)
+   - If you want to keep your existing database and update (e.g. after a rebase)
+     - If there are new migrations, run `make migrate`. (Or, because we're pre-production, you can delete the existing migrations, run `make migrations` and then `make migrate`)
+     - If there are new fixtures, `run python manage.py loaddata <path-to-fixture>``
+3. Run `make run`
+
+### Troubleshooting
+
+- There might be an error where postgres is only installed with the `postgres` user and not your local user.
+  To address this, in a terminal, enter:
+
+```bash
+$> sudo -u postgres createuser -s -i -d -r -l -w <<your_local_user>>
+$> sudo -u postgres createdb <<your_local_user>>
+```
+
+- if running poetry commands throws errors about the `_hashlib` library, [try these troubleshooting commands](https://github.com/python-poetry/poetry/issues/7695#issuecomment-1572825140)
+- if you encounter an issue installing `psycopg2` relating to `Error: pg_config --libdir failed: No version is set for command pg_config`, try setting your Postgres version for asdf globally rather than locally. (E.g., `asdf global postgres 14.0`)
+- if your admin panel looks ugly, run `python3 manage.py collectstatic` to collect static files and then try again.
+
+## Frontend Environment Setup
+
+In the `client` directory:
+
+1. To install depedencies: `yarn`
+2. To run the development server: `yarn dev`
+Open [http://localhost:3000](http://localhost:3000) with your browser to see the result.
+You can start editing the page by modifying `pages/index.tsx`. The page auto-updates as you edit the file.
+
+## Pre-Commit
+
+TODO: Add pre-commit
```

Comment:
note: precommit CI stuff is awaiting review here https://github.com/orgs/bcgov/projects/123/views/2?pane=issue&itemId=49861453